### PR TITLE
Add residue ring of local fields

### PR DESCRIPTION
**Pull request by joschmitt**

### CODE REVIEW

File: test/LocalField.jl [hunk 1]
```diff
@@ -4,4 +4,5 @@
   include("LocalField/LocalField.jl")
   include("LocalField/neq.jl")
   include("LocalField/Completions.jl")
+  include("LocalField/ResidueRing.l")
```
Comment (thofma) [inline test/LocalField.jl:7]:
```suggestion
  include("LocalField/ResidueRing.jl")
```

File: src/LocalField/ResidueRing.jl [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+  @req parent(a) === parent(b) "Parents do not match"
+  @req !is_zero(b) "Division by 0"
+  if is_zero(a)
+    return zero(parent(a))
+  end
+  @req valuation(data(a)) >= valuation(data(b)) "Division not possible"
+  c = divexact(data(a), data(b))
+  setprecision!(c, _exponent(parent(a)))
+  return parent(a)(c, copy = false, check = false)
+end
```
Comment (thofma) [inline src/LocalField/ResidueRing.jl:251]: @fieker Can you confirm that this makes sense?

**Comment by thofma** [comment]
Splendid! Can you hook up the solving/kernel to use the Howell form?

**Comment by fieker** [comment]
the precision of the zero is wrong, I think. It should(?) depend on the precision of a and b, not of the parent...

File: src/LocalField/ResidueRing.jl [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+
+################################################################################
+#
+#  Basic functionality
+#
+################################################################################
+
+characteristic(R::LocalFieldValuationRingResidueRing) = prime(_field(R))^_exponent(R)
+
+zero(R::LocalFieldValuationRingResidueRing) = R()
```
Comment (fieker) [inline src/LocalField/ResidueRing.jl:37]: unless the zerois exact, it needs a precision....

Comment (thofma) [reply src/LocalField/ResidueRing.jl:37]: This is `Zp/p^k`, so the zero is the exact zero of this ring.

Comment (joschmitt) [reply src/LocalField/ResidueRing.jl:37]: I mean, this one is exact because it is the zero of the residue ring?

**Review by fieker — APPROVED** [review]
I'm not 100% sure all the precision is correct, but otherwise it is looking good

**Pull request merged**